src/pages/DetailedPage.jsx

```diff
@@ -29,74 +44,102 @@ export default function DetailedPage(props) {
   };
 
   useEffect(() => {
-    (async function () {
-      await axios
-        .post('http://localhost:3000/get-location-current-weather-by-id', {
-          id: locationId,
-          temp: temperature
-        })
-        .then(data => setLocation(data.data))
-        .catch(err => {
-          console.error('error occured: ', err.message);
-        });
-    })();
+    if (locationId) {
+      (async function () {
+        await axios
+          .post('http://localhost:3000/get-location-current-weather-by-id', {
```

Comment:
seems like this request call can be moved to separate function, it was used twice in one component

Reply:
see no updates about this comment